src/main/java/org/datadog/jenkins/plugins/datadog/DatadogBuildListener.java

```diff
@@ -134,7 +133,14 @@ public final void onStarted(final Run run, final TaskListener listener) {
 
       BuildStartedEventImpl evt = new BuildStartedEventImpl(builddata, tags);
       DatadogHttpRequests.sendEvent(evt);
-      gauge("jenkins.job.waiting", builddata, "waiting", extraTags);
+
+      // Send waiting metric if item.getInQueueSince() is available
+      try {
+        builddata.put("waiting", (System.currentTimeMillis() - item.getInQueueSince()) / DatadogBuildListener.THOUSAND_LONG);
+        gauge("jenkins.job.waiting", builddata, "waiting", extraTags);
+      } catch (NullPointerException e) {
+        logger.severe(e.getMessage());
```

Comment:
Is it really a severe condition? We don't want to be overly verbose if this is something we might run in to habitually.